cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -109,6 +110,11 @@ func ResyncPeriod(s *options.CMServer) func() time.Duration {
 
 // Run runs the CMServer.  This should never exit.
 func Run(s *options.CMServer) error {
+	// To help debugging, immediately log version and command line flags.
+	glog.Infof("%+v", version.Get())
```

Comment:
TBH, the codebase has shifted so much that I no longer know where is the sanest place to put utilities like that, I'd have to re-re-rationalize it with the shifting.